### ISSUE TITLE
[Bug] GetPropertyType incorrectly holds a u8 instead of a u32

### ISSUE THREAD

**Issue by Jeansidharta** [bread-graphics/breadx#92]
# My Specific case

I'm writing a program that reads the current focused window title. For that, I'm trying to read the `WM_NAME` property of my focused window. By using the `xprop` command, I can see the atom type of the `WM_NAME` is `UTF8_STRING`, and by using the command `xlsatoms | grep UTF8_STRING`, I can see it's `id` is `408`. Therefore, I tried writing this code:

```rust
const UTF8_STRING_ATOM: u32 = 408;
let window_title = {
    let cookie = connection.get_property(
        false,
        focused_window,
        wm_name_atom,
        UTF8_STRING_ATOM,
        0,
        1024,
    )?;
    connection.wait_for_reply(cookie)?
};
```
However, I'm getting this compiler error:
```
UTF8_STRING_ATOM,
^^^^^^^^^^^^^^^^ the trait `From<u32>` is not implemented for `GetPropertyType`
```

# The Problem

The struct [GetPropertyType](https://docs.rs/breadx/3.1.0/breadx/protocol/xproto/struct.GetPropertyType.html#) internally holds a `u8` instead of a `u32`, and therefore, when calling [connection.get_property](https://docs.rs/breadx/3.1.0/breadx/display/trait.DisplayFunctionsExt.html#method.get_property), the atom type can only be up to 255.

This does not conform to the [X11 specification](https://www.x.org/releases/current/doc/xproto/x11protocol.html#requests:GetProperty), which states that the `type` argument is of type `ATOM`, which is a [32 bit value](https://www.x.org/releases/current/doc/xproto/x11protocol.html#Common_Types).

# Proposed Solution

Change the internal value of `GetPropertyType` to be a u32, and implement the trait `From<u32>`.

# Workarounds

Instead of using the correct type in the function `get_property`, using the `breadx::protocol::xproto::AtomEnum::ANY` type works. Example:

```rust

let any_type_atom: u8 = breadx::protocol::xproto::AtomEnum::ANY.into();
let window_title = {
    let cookie = connection.get_property(
        false,
        focused_window,
        wm_name_atom,
        any_type_atom,
        0,
        1024,
    )?;
    connection.wait_for_reply(cookie)?
};
```

**Comment by Jeansidharta** [comment]
I just realized the `GetPropertyType` is not defined in this crate. It's defined in the [x11rb](https://docs.rs/x11rb/latest/x11rb/protocol/xproto/struct.GetPropertyType.html) crate. However, the `GetPropertyType` struct doesn't seem to be used anywhere in the x11rb crate.

In x11rb, the `get_property` function is defined [here](https://docs.rs/x11rb/0.12.0/x11rb/protocol/xproto/trait.ConnectionExt.html#method.get_property), and the `GetPropertyRequest` struct is defined [here](https://docs.rs/x11rb/0.12.0/x11rb/protocol/xproto/struct.GetPropertyRequest.html). In both instances, the type can be supplied with a `u32`.

Maybe the solution then would be to not use the `GetPropertyType` struct at all, and replace all instances of `impl Into<GetPropertyType>` (such as in [here](https://docs.rs/breadx/3.1.0/breadx/display/trait.DisplayFunctionsExt.html#method.get_property)) with `impl Into<Atom>`?

**Comment by notgull** [comment]
Yeah, using a less strict option would probably work in this case.
